server/protocols/webadmin/src/test/java/org/apache/james/webadmin/routes/DomainRoutesTest.java

```diff
@@ -0,0 +1,341 @@
+                .contentType(ContentType.JSON)
+            .when()
+                .post(DomainRoutes.DOMAIN + DOMAIN)
+            .then()
+                .statusCode(200);
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(DomainRoutes.DOMAIN + DOMAIN)
+            .then()
+                .statusCode(200);
+        }
+
+        @Test
+        public void getDomainSHouldReturnNotFoundWhenTheDomainIsAbsent() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(DomainRoutes.DOMAIN + DOMAIN)
+            .then()
+                .statusCode(404);
+        }
+
+    }
+
+    public class ExceptionHandling {
+
+        private DomainList domainList;
+        private String domain;
+
+        @Before
+        public void setUp() throws Exception {
+            domainList = mock(DomainList.class);
+            createServer(domainList);
+            domain = "domain";
+        }
+
+        @Test
+        public void deleteShouldReturnErrorOnUnknownException() throws Exception {
+            doThrow(new RuntimeException()).when(domainList).removeDomain(domain);
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .delete(DomainRoutes.DOMAIN + "domain")
+            .then()
+                .statusCode(500);
+        }
+
+        @Test
+        public void postShouldReturnErrorOnUnknownException() throws Exception {
+            doThrow(new RuntimeException()).when(domainList).addDomain(domain);
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .post(DomainRoutes.DOMAIN + "domain")
+            .then()
+                .statusCode(500);
+        }
+
+        @Test
+        public void getDomainShouldReturnErrorOnUnknownException() throws Exception {
+            when(domainList.containsDomain(domain)).thenThrow(new RuntimeException());
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(DomainRoutes.DOMAIN + "domain")
+            .then()
+                .statusCode(500);
+        }
+
+        @Test
+        public void getDomainsShouldReturnErrorOnUnknownException() throws Exception {
+            when(domainList.getDomains()).thenThrow(new RuntimeException());
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(DomainRoutes.DOMAINS)
+            .then()
+                .statusCode(500);
+        }
+
+        @Test
+        public void deleteShouldReturnOkWhenDomainListException() throws Exception {
```

Comment:
I first return a 4xx error code.

When @mbaechler read the README specifying the spec of the API I got a : 

```
> +Response codes :
> + - 200 : The user was successfully deleted
> + - 400 : The URL do not match the specification
> + - 406 : The user did not exist

and ?
```

Other examples includes delete on user and domain (identical to this one).

I got a fancy : 

```
> +Response codes :
> + - 200 : The domain was successfully added
> + - 400 : The URL do not match the specification
> + - 406 : The domain already exist

and so what ?
```

When working on domain creation.

As such I understood : **I don't care about your response code, just about the result**.

So, clearly  the team disagree on this. Hoppefully, PR are a dreamed moment to solve such disagreements.

My point is : I overall agree with @mbaechler . Not enforcing these response code is a good idea as it is less tied to the backend. Such conciderations are for instance advised for scalable API (https://blog.acolyer.org/2015/04/24/the-scalable-commutativity-rule-designing-scalable-software-for-multicore-processors/). And what would we do with such response codes?

So what do you think ?